flink-streaming-java/src/main/java/org/apache/flink/streaming/api/datastream/SinkOperatorsUidHashes.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.api.datastream;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.annotation.PublicEvolving;
+
+import javax.annotation.Nullable;
+
+/**
+ * This class is responsible to hold operator Uid hashes from the common operators of the sink. With
+ * this, users can recover a sink snapshot that did not bind uids to the operator before changing
+ * the topology.
+ */
+@PublicEvolving
+public class SinkOperatorsUidHashes {
```

Comment:
I'd call it `CustomSinkOperatorUidHashes`. It should be clear that this is purely optional.